lib/Doctrine/Common/Cache/ApcCache.php

```diff
@@ -85,6 +85,20 @@ protected function doFetchMultiple(array $keys)
     /**
      * {@inheritdoc}
      */
+    protected function doSaveMultiple(array $keysAndValues, $lifetime = 0)
+    {
+        $result = apc_store($keysAndValues, null, $lifetime);
+
+        if ($result === false || count($result)) {
```

Comment:
Checking against `false` seems wrong, as you will always get an `array` result, no?
Also, you should probably just use `empty()`, to avoid counting the elements when not needed.